src/press/views.py

```diff
@@ -80,19 +81,19 @@ def sitemap(request):
     :param request: HttpRequest object
     :return: HttpResponse object
     """
-    try:
-        if request.journal is not None:
-            # if there's a journal, then we render the _journal_ sitemap, not the press
-            return journal_views.sitemap(request)
+    if request.journal is not None:
+        # if there's a journal, then we render the _journal_ sitemap, not the press
+        return journal_views.sitemap(request)
 
-        if request.repository is not None:
-            # if there is a repository we return the repository sitemap.
-            return repository_views.repository_sitemap(request)
+    if request.repository is not None:
+        # if there is a repository we return the repository sitemap.
+        return repository_views.sitemap(request)
 
-        return files.serve_sitemap_file(['sitemap.xml'])
-    except FileNotFoundError:
-        logger.warning('Sitemap for {} not found.'.format(request.press.name))
-        raise Http404()
+
+    return core_views.sitemap(
+        request,
+        ['sitemap.xml'],
+    )
```

Comment:
I'm thinking it makes more sense for core to  redirect to the correct site type of press/journal/repository, to avoid the interdependence between the. That said, it does not currently present a circular import and `Press` already knows about `journal` and `repository`, so I'll leave this as an optional comment